### PR TITLE
enhancement/use-libc-api-to-get-all-interfaces)

### PR DESCRIPTION
**Pull request by xiar**
Use libc API getifaddrs to get all interfaces.

### CODE REVIEW

**Comment by emcinfrasim** [comment]
Can one of the admins verify this patch?

**Comment by xiar** [comment]
@fub2 @turtle-fly @XiaowenJiang Please review this PR. thanks.

File: infrasim/helper.py [hunk 1]
```diff
@@ -53,31 +60,130 @@ def get_interface_ip(ifname):
...
+            name = ifa.ifa_name
+            i = retval.get(name)
+            if not i:
+                i = retval[name] = NetworkInterface(name)
+            family, addr = getfamaddr(ifa.ifa_addr.contents)
+            if addr:
+                i.addresses[family] = addr
+        return retval.values()
+    finally:
+        libc.freeifaddrs(ifap)
```
Comment (XiaowenJiang) [inline infrasim/helper.py]: cool

**Comment by XiaowenJiang** [comment]
retest this please

**Comment by turtle-fly** [comment]
:+1:

**Pull request merged**